### DISCUSSION TITLE
DGS Codegen- custom annotations over class

### DISCUSSION THREAD

**Discussion by korbecki** [Netflix/dgs-framework#1848] [Ideas]
Hi, what do you think about the idea of adding to the generator the ability to place annotations over a class, for example I would like to add the @FieldNameConstants annotation. There is similar functionality in OpenAPI Generator

**Comment by paulbakker** [comment]
Our general thinking is that when customization of generated types is necessary, you're probably better off creating your own types instead of relying on code gen. It's very common to have multiple Java classes represent the same "thing", but with different concerns:
1. The input and result types of datafetchers (this is what CodeGen is for)
2. Entities for JPA
3. gRPC types generated from Proto
4. ...

It's tempting to want to re-use the same (generated) class for all these concerns, because they're _mostly_ the same. This leads to mixing concerns that shouldn't be mixed, and making compromises in, for example, what fields are exposed.
Specially with Records now available in Java, it's really easy to create new types for just a single purpose, which gives better decoupling, without it being a lot of work.

The reason is that such customizations, even if they seem simple, very quickly get out of hand. This makes both CodeGen more complicated and makes your code base harder to reason about.

For your specific case, `@FieldNameConstants` is a Lombok annotation, so that's another level of generation. That complicates things further, and I would recommend against it.
